sway/config.c

```diff
@@ -130,6 +132,17 @@ static void destroy_removed_seats(struct sway_config *old_config,
 	}
 }
 
+int get_font_text_height(char *font) {
+	cairo_t *cairo = cairo_create(NULL);
+	int text_height;
+	get_text_size(cairo, font, NULL, &text_height, 1, false,
+			"ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+			"abcdefghijklmnopqrstuvwxyz"
+			"!@#$%^&*([{|");
```

Comment:
This approach isn't going to cut it, you'll have to render the text and if it's taller/shorter than you expected you'll need to re-compute the border geometry and re-render. Re-render compositor-wide if this happens, it would be weird to see each window with a differently sized border.